ale_linters/ansible/ansible_lint.vim

```diff
@@ -51,6 +39,6 @@ call ale#linter#Define('ansible', {
 \   'name': 'ansible_lint',
 \   'aliases': ['ansible', 'ansible-lint'],
 \   'executable': function('ale_linters#ansible#ansible_lint#GetExecutable'),
-\   'command': '%e -p %t',
+\   'command': '%e -p -x yaml',
```

Comment:
Find out what the earliest version of `ansible-lint` is that supports the `-x` option. We need to make sure it's old enough that we won't break linting for users who are stuck on older versions. About 1-2 years ago is usually old enough.

Reply:
_ansible-lint_ version 4.1.0, which is around 2 years old, already supports -x option.

Autodetection is only supported from version 4.3.0. And just from version 5.0.0, which is around half a month old, it's actually working as expected.
 
Working as expected means that if any _.yaml_ file in a _git_ repo of ansible project is opened and correctly identified as _ansible_ filetype, the linter will correctly identify the associated playbook, report all findings for all the involved tasks, roles, etc.. PR ale _ansible-lint_ definition code will filter only the ones appropriate for the opened file. Which is the expected behavior as I understand.

All previous versions had to get the playbook's absolute path as an argument. The findings won't be shown if another ansible file would be opened, e.g. some tasks file.

So my suggested solution works only from version 5.0.0. But it works as expected of any other linter - linting errors are returned for any opened ansible file in a project. And not only for playbook.

The current implementation doesn't work at all - it tries to lint temporary file, so all referenced files in a playbook are not being linted and reported. If one opens a task file, nothing will be reported at all, because it's not a playbook file.

Reply:
Can confirm the -x option exists since 4.0.0 that is from 2018. I think is safe to assume the option is available for most users by now.